commands/operator-sdk/cmd/scorecard/olm_tests.go

```diff
@@ -148,6 +176,79 @@ func crdsHaveResources(csv *olmapiv1alpha1.ClusterServiceVersion) {
 	}
 }
 
+func getUsedResources() ([]schema.GroupVersionKind, error) {
+	logs, err := getProxyLogs()
+	if err != nil {
+		return nil, err
+	}
+	var resources []schema.GroupVersionKind
+	for _, line := range strings.Split(logs, "\n") {
+		logMap := make(map[string]interface{})
+		err := json.Unmarshal([]byte(line), &logMap)
+		if err != nil {
+			// it is very common to get "unexpected end of JSON input", so we'll leave this at the debug level
+			log.Debugf("could not unmarshal line: %v", err)
+			continue
+		}
+		/*
+			There are 6 formats a resource uri can have:
+			Cluster-Scoped:
+				Collection: /apis/GROUP/VERSION/KIND
+				Individual: /apis/GROUP/VERSION/KIND/NAME
+				Core:       /api/v1/KIND
+			Namespaces:
+				All Namespaces:          /apis/GROUP/VERSION/KIND (same as cluster collection)
+				Collection in Namespace: /apis/GROUP/VERSION/namespaces/NAMESPACE/KIND
+				Individual:              /apis/GROUP/VERSION/namespaces/NAMESPACE/KIND/NAME
+				Core:                    /api/v1/namespaces/NAMESPACE/KIND
+
+			These urls are also often appended with options, which are denoted by the '?' symbol
+		*/
+		if msg, ok := logMap["msg"].(string); !ok || msg != "Request Info" {
+			continue
+		}
+		uri, ok := logMap["uri"].(string)
+		if !ok {
+			log.Warn("URI type is not string")
+			continue
+		}
+		removedOptions := strings.Split(uri, "?")[0]
+		splitURI := strings.Split(removedOptions, "/")
+		// first string is empty string ""
+		splitURI = splitURI[1:]
+		switch len(splitURI) {
+		case 3:
+			if splitURI[0] == "api" {
+				resources = append(resources, schema.GroupVersionKind{Version: splitURI[1], Kind: splitURI[2]})
+			}
+		case 4:
+			if splitURI[0] == "apis" {
+				resources = append(resources, schema.GroupVersionKind{Group: splitURI[1], Version: splitURI[2], Kind: splitURI[3]})
+			}
+		case 5:
+			if splitURI[0] == "api" {
+				resources = append(resources, schema.GroupVersionKind{Version: splitURI[1], Kind: splitURI[4]})
+			} else if splitURI[0] == "apis" {
+				resources = append(resources, schema.GroupVersionKind{Group: splitURI[1], Version: splitURI[2], Kind: splitURI[3]})
+			}
+		case 6, 7:
+			if splitURI[0] == "apis" {
+				resources = append(resources, schema.GroupVersionKind{Group: splitURI[1], Version: splitURI[2], Kind: splitURI[5]})
+			}
+		}
+	}
+	// remove duplicates
+	addedResources := map[schema.GroupVersionKind]bool{}
+	var deduplicatedResources []schema.GroupVersionKind
+	for _, resource := range resources {
```

Comment:
Nit: It might be a little simpler to make `resources` a `map[schema.GroupVersionKind]bool` so that they're already deduplicated when the loop completes. Then just range over the map to build the slice that we return.